src/dnf-plugins/product-id/product-id.c

```diff
@@ -308,24 +314,13 @@ int pluginHook(PluginHandle *handle, PluginHookId id, DnfPluginHookData *hookDat
             if (tmp_err) {
                 printError("Unable to get information about repository", tmp_err);
             } else if (repoMd != NULL) {
-                LrYumRepoMdRecord *repoMdRecord = lr_yum_repomd_get_record(repoMd, "productid");
-                if (repoMdRecord) {
+                RepoProductId *repoProductId = initRepoProductId();
+                int fetchSuccess = isProductIdDownloaded(repo, repoProductId);
+                if (fetchSuccess) {
                     debug("Repository %s has a productid", dnf_repo_get_id(repo));
-                    RepoProductId *repoProductId = initRepoProductId();
-                    // TODO: do not fetch productid certificate, when dnf context is set to cache-only mode
-                    // Microdnf nor PackageKit do not support this feature ATM
```

Comment:
Is the cache-only mode no longer a concern?

Reply:
I will talk about this more with dnf guys, but I believe that it isn't our concern anymore, because we request downloading product-id certificate using libdnf API and it's saved in cache of Microdnf or PackageKit (e.g. `/var/cache/PackageKit/29/metadata/content-label-no-gpg-29-x86_64/repodata/`).